tiatoolbox/models/engine/engine_abc.py

```diff
@@ -1,21 +1,610 @@
+            Path.cwd()
+        )  # save the output to current working directory and return save_dir
+
+    if len_images > 1:
+        logger.info(
+            "When providing multiple whole slide images, "
+            "the outputs will be saved and the locations of outputs "
+            "will be returned to the calling function.",
+        )
+
+    save_dir = Path(save_dir)
+    save_dir.mkdir(parents=True, exist_ok=overwrite)
+
+    return save_dir
 
 
 class EngineABC(ABC):
-    """Abstract base class for engines used in tiatoolbox."""
+    """Abstract base class for engines used in tiatoolbox.
+
+    Args:
+        model (str | nn.Module):
+            A PyTorch model. Default is `None`.
+            The user can request pretrained models from the toolbox using
+            the list of pretrained models available at this `link
+            <https://tia-toolbox.readthedocs.io/en/latest/pretrained.html>`_
+            By default, the corresponding pretrained weights will also
+            be downloaded. However, you can override with your own set
+            of weights.
+        weights (str or Path):
+            Path to the weight of the corresponding `model`.
+
+            >>> engine = EngineABC(
+            ...    pretrained_model="pretrained-model-name",
+            ...    weights="pretrained-local-weights.pth")
+
+        batch_size (int):
+            Number of images fed into the model each time.
+        num_loader_workers (int):
+            Number of workers to load the data using :class:`torch.utils.data.Dataset`.
+            Please note that they will also perform preprocessing. default = 0
+        num_post_proc_workers (int):
+            Number of workers to postprocess the results of the model. default = 0
+        device (str):
+            Select the device to run the model. Default is "cpu".
+        verbose (bool):
+            Whether to output logging information.
+
+    Attributes:
+        images (str or :obj:`pathlib.Path` or :obj:`numpy.ndarray`):
+            A NHWC image or a path to WSI.
+        patch_mode (str):
+            Whether to treat input image as a patch or WSI.
+            default = True.
+        model (str | nn.Module):
+            Defined PyTorch model.
+            Name of the existing models supported by the TIAToolbox for
+            processing the data. For a full list of pretrained models,
+            refer to the `docs
+            <https://tia-toolbox.readthedocs.io/en/latest/pretrained.html>`_
+            By default, the corresponding pretrained weights will also
+            be downloaded. However, you can override with your own set
+            of weights via the `weights` argument. Argument
+            is case-insensitive.
+        ioconfig (ModelIOConfigABC):
+            Input IO configuration to run the Engine.
+        _ioconfig ():
+            Runtime ioconfig.
+        return_labels (bool):
+            Whether to return the labels with the predictions.
```

Comment:
I think this could be more clear. What does it mean by labels? What is the format of the output when this is true versus false?

Reply:
This flag is False by default and is implicitly set to true if the engine_abc's `run()` method includes an optional labels list. A relevant example is now added.